### PR TITLE
Remove downlevel runtimes from SiteExtensions package

### PR DESCRIPTION
**Pull request by wtgodbe**
We're getting more and more security-related warnings about these out-of-support runtimes. At this point it's probably more trouble than it's worth to keep bundling them in the SiteExtension package.

### CODE REVIEW

**Comment by wtgodbe** [comment]
CC @danmoseley this should clear up some of the BinSkim errors we've been seeing

File: eng/Dependencies.props [hunk 1]
```diff
@@ -141,15 +141,8 @@ and are generated based on the last package release.
 
   <ItemGroup Label=".NET team dependencies (Non-source-build)" Condition="'$(DotNetBuildSourceOnly)' != 'true'">
     <LatestPackageReference Include="Microsoft.AspNet.WebApi.Client" />
-    <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.2.1" />
```
Comment (BrennanConroy) [inline eng/Dependencies.props:144]: Should 2.1 be removed?

**Review by BrennanConroy — APPROVED** [review]

File: eng/Dependencies.props (hunk 1 above)
Comment (wtgodbe) [reply eng/Dependencies.props:144]: I think so, it contains only `netcoreapp2.1` assets, which isn't a supported TFM anymore. Plus it's not on the list of supported packages: https://dotnet.microsoft.com/en-us/platform/support/policy/aspnet/2.1-packages

File: eng/Dependencies.props [hunk 2]
```diff
@@ -141,15 +141,8 @@ and are generated based on the last package release.
...
-    <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.3.1.x64" />
-    <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.3.1.x86" />
-    <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.5.0.x64" />
-    <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.5.0.x86" />
     <LatestPackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.6.0.x64" />
```
Comment (danmoseley) [inline eng/Dependencies.props:144]: do we need 6.0? It's about to be out of support.

Comment (wtgodbe) [reply eng/Dependencies.props:144]: It's still in support for a few months, I don't think we should remove it until it's officially out of support

Comment (danmoseley) [reply eng/Dependencies.props:144]: might be worth something on the backlog to do that else it will just show up later in s360

**Comment by danmoseley** [comment]
Do we need any manual step to get the updated package published/code flowed?

File: eng/Dependencies.props (hunk 2 above)
Comment (wtgodbe) [reply eng/Dependencies.props:144]: This now falls under general 6.0 cleanup that I'll do once that's out of support, which I'm tracking w/ https://github.com/dotnet/aspnetcore-internal/issues/4591

**Comment by wtgodbe** [comment]
> Do we need any manual step to get the updated package published/code flowed?

Nope, it gets published automatically each release, so the preview7 package will be the first without the out of support downlevel TFMs.

**Pull request merged**